halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Node.java

```diff
@@ -414,53 +414,101 @@ public NodeDiff diff(Node other) {
+    return fPath;
+  }
 
-    recursiveConsume(fileFinder);
+  /**
+   * Changes the file path pointed by "localFile" and beginning with "rootPath" from being absolute,
+   * to a relative path by removing "rootPath".
+   *
+   * @param localFile the field annotated with {@see LocalFile} to transform.
+   * @param rootPath prefix to remove from the local path.
+   * @return original absolute file path, or null if the file path was not modified.
+   */
+  public String makeAbsolutePathRelative(Field localFile, String rootPath) {
+    String fPath = getStringFieldValue(localFile);
+    if (StringUtils.isEmpty(fPath)) {
+      return null;
+    }
+    Path relativePath = toRelativePath(rootPath, fPath);
+    setStringFieldValue(localFile, relativePath.toString());
+    return fPath;
+  }
+
+  public String getStringFieldValue(Field f) {
+    try {
+      f.setAccessible(true);
+      String value = (String) f.get(this);
+      if (StringUtils.isNotEmpty(value)) {
+        return value;
+      }
+      // if no value in field, try using getter
+      try {
+        return (String)
+            this.getClass().getMethod("get" + StringUtils.capitalize(f.getName())).invoke(this);
+      } catch (NoSuchMethodException | InvocationTargetException ignored) {
+        return null;
+      }
+    } catch (IllegalAccessException e) {
+      return null;
+    } finally {
+      f.setAccessible(false);
+    }
   }
 
-  public void makeLocalFilesRelative(String halconfigPath) {
-    swapLocalFilePrefixes(LocalFile.RELATIVE_PATH_PLACEHOLDER, halconfigPath);
+  public void setStringFieldValue(Field f, String value) {
+    try {
+      f.setAccessible(true);
+      f.set(this, value);
+    } catch (IllegalAccessException e) {
+      throw new HalException(
+          FATAL, String.format("Unable to set value %s to field %s", value, f.getName()));
+    } finally {
+      f.setAccessible(false);
+    }
   }
 
-  public void makeLocalFilesAbsolute(String halconfigPath) {
-    swapLocalFilePrefixes(halconfigPath, LocalFile.RELATIVE_PATH_PLACEHOLDER);
+  private Path toAbsolutePath(String rootPath, String relativePath) {
+    Path absolutePath = Paths.get(rootPath, relativePath).normalize();
+    if (!absolutePath.startsWith(rootPath)) {
+      throw new HalException(
+          FATAL,
+          "Error resolving file path '"
+              + relativePath
+              + "': Relative file paths must resolve to files inside "
+              + rootPath);
+    }
+    return absolutePath;
+  }
+
+  private Path toRelativePath(String rootPath, String absolutePath) {
+    if (!absolutePath.startsWith(rootPath)) {
+      throw new HalException(
+          FATAL,
+          "Local file: " + absolutePath + " has incorrect prefix - must begin with " + rootPath);
+    }
+    if (!rootPath.endsWith(File.separator)) {
+      rootPath += File.separator;
+    }
+    return Paths.get(absolutePath.substring(rootPath.length()));
```

Comment:
`toRelativePath('/foo', '/foo2/bar')` returns `/bar`.

Don't try to do path resolution logic yourself; it's more complicated than you think. Convert these `Strings` to `Path` objects and use the built-in methods like `relativize`.